web/src/transmission.js

```diff
@@ -373,10 +373,21 @@ export class Transmission extends EventTarget {
         break;
 
       case Prefs.RefreshRate: {
-        clearInterval(this.refreshTorrentsInterval);
         const callback = this.refreshTorrents.bind(this);
```

Comment:
Putting PR 6998 aside, if we are only considering performance, then maybe something like this is better.

```suggestion
        const callback = () => {
          if (!document.hidden) {
            this.refreshTorrents();
          }
        };
```

I don't like the sound of calling `Date.now()` on every monitor refresh... I don't know how it's like in JS, but reading the clock is generally an expensive operation in any language.

Reply:
The Date.now() became redundant as elapsed time measurement aren't needed as way to control FPS because setTimeout already got the time measurement in hand. requestAnimationFrame takes the role of interval and setTimeout to control length BFF 🤍

Reply:
I'm wary of page with old info when adding refresh timers with abilities to pause in background, document.hidden is no exception, I've played with document.hidden for a bit, my impression is: it will maintain old page for a few seconds (seconds left of the setTimeout) when I switch tabs imminently so I'm not incorporating it this time. Thank you for sharing performance tips.